README.md
# 🎈 ClickMe 🎈

### A react library full of highly customizable and animated buttons out of the box

![Preview of a few available animations](public/preview.gif)

## Quick installation and usage

- NPM installation

```bash
npm i react-clickme
```

- Importing in a React component

```javascript
import ClickMe from "./components/global/clickmeButton";

export default function App() {
  return <ClickMe>Example text</ClickMe>;
}
```

- Customization

```javascript
import ClickMe from "./components/global/clickmeButton";

const clickmeProps = {
  ariaLabel: "my-btn",
  style: { fontWeight: "500", padding: "0.5rem 1rem" },
  primaryBgColor: "#3D7DF6",
  primaryTextColor: "#fff",
  secondaryBgColor: "#000",
  hoverType: "slideToRight",
  idleType: "pulse",
};

export default function App() {
  return <ClickMe {...clickmeProps}>Example text</ClickMe>;
}
```

## Documentation

- [Docs on the official website](https://react-clickme.vercel.app/docs)
- [Installation](docs/installation.md)
- [Usage/Examples](docs/usage.md)

# To see all of the available custom options and presets, click <a href="https://react-clickme.vercel.app" target="_blank" rel="noreferrer">here</a>. 👈

<img width="300" src="public/logo.svg" alt="ClickMe logo" />


CHANGELOG.md
# 1.0.0 (2023-12-05)


### Features

* initial commit ([70e4c36](https://github.com/MuchaSsak/clickme-website/commit/70e4c36753e3fd4aa8cc594468e423a46d185e91))


### BREAKING CHANGES

* initial commit





data/clickmeButtons.ts
export default [
  {
    category: "hoverType",
    name: "shutterInHorizontal",
    id: crypto.randomUUID(),
  },
  {
    category: "hoverType",
    name: "shutterOutHorizontal",
    id: crypto.randomUUID(),
  },
  {
    category: "hoverType",
    name: "shutterInVertical",
    id: crypto.randomUUID(),
  },
  {
    category: "hoverType",
    name: "shutterOutVertical",
    id: crypto.randomUUID(),
  },
  { category: "hoverType", name: "sweepToTop", id: crypto.randomUUID() },
  { category: "hoverType", name: "sweepToRight", id: crypto.randomUUID() },
  { category: "hoverType", name: "sweepToBottom", id: crypto.randomUUID() },
  { category: "hoverType", name: "sweepToLeft", id: crypto.randomUUID() },
  {
    category: "hoverType",
    name: "rectangleIn",
    id: crypto.randomUUID(),
  },
  {
    category: "hoverType",
    name: "rectangleOut",
    id: crypto.randomUUID(),
  },
  { category: "hoverType", name: "slideToRight", id: crypto.randomUUID() },
  { category: "hoverType", name: "slideToLeft", id: crypto.randomUUID() },
  { category: "hoverType", name: "radialOut", id: crypto.randomUUID() },
  { category: "hoverType", name: "changeColor", id: crypto.randomUUID() },
  {
    category: "hoverType",
    name: "increaseLetterSpacing",
    id: crypto.randomUUID(),
  },
  { category: "hoverType", name: "invert", id: crypto.randomUUID() },
  { category: "hoverType", name: "scaleUp", id: crypto.randomUUID() },
  { category: "hoverType", name: "scaleDown", id: crypto.randomUUID() },
  { category: "hoverType", name: "outline", id: crypto.randomUUID() },
  { category: "hoverType", name: "translateUp", id: crypto.randomUUID() },
  {
    category: "hoverType",
    name: "increaseBrightness",
    id: crypto.randomUUID(),
  },
  {
    category: "hoverType",
    name: "decreaseBrightness",
    id: crypto.randomUUID(),
  },
  { category: "idleType", name: "pulse", id: crypto.randomUUID() },
  { category: "idleType", name: "doublePulse", id: crypto.randomUUID() },
  { category: "idleType", name: "shadowPulse", id: crypto.randomUUID() },
  { category: "idleType", name: "bounce", id: crypto.randomUUID() },
  { category: "idleType", name: "tada", id: crypto.randomUUID() },
  { category: "idleType", name: "translateUp", id: crypto.randomUUID() },
  { category: "clickType", name: "rubberBand", id: crypto.randomUUID() },
  { category: "clickType", name: "headShake", id: crypto.randomUUID() },
  { category: "clickType", name: "wobble", id: crypto.randomUUID() },
  { category: "clickType", name: "jello", id: crypto.randomUUID() },
];


data/docsUsages.ts
export default [
  {
    name: "Importing the component",
    code: `import ClickMe from "react-clickme";

export default function App() {
   return <ClickMe>Example text</ClickMe>;
}`,
  },
  {
    name: "Passing in custom props along with regular ones for JSX button element",
    code: `import ClickMe from "react-clickme";

export default function App() {
   return (
      <ClickMe
         className="btn"
         hoverType="changeColor"
         aria-label="my-btn"
         onClick={() => alert("Hello")}
      >
         hello
      </ClickMe>
   );
}`,
  },

  {
    name: "Destructuring props trick for cleaner JSX",
    code: `import ClickMe from "react-clickme";

const clickmeProps = {
   className: "btn",
   hoverType: "rectangleOut",
   ariaLabel: "my-btn",
   onClick: () => alert("Hello"),
};

export default function App() {
   return <ClickMe {...clickmeProps}>Example text</ClickMe>;
}`,
  },
  {
    name: "Custom inline CSS styles",
    code: `import ClickMe from "react-clickme";

const clickmeStyles = {
   fontSize: "2rem",
   fontWeight: "500",
   padding: "1rem 0.5rem",
};

export default function App() {
   return <ClickMe style={clickmeStyles}>Example text</ClickMe>;
}`,
  },
  {
    name: "You can using multiple animation types!",
    code: `import ClickMe from "react-clickme";

export default function App() {
   const clickmeProps = {
      hoverType: "changeColor",
      idleType: "pulse",
      clickType: "jello",
   };

   return <ClickMe {...clickmeProps}>Example text</ClickMe>;
}`,
  },
  {
    name: "Example of consistant buttons across the entire app",
    code: `
// ----- someFile.js -----
export const CLICKME_PRIMARY = {
   primaryBgColor: "#3D90F6",
   primaryTextColor: "#fff",

   idleType: "doublePulse",
};

export const CLICKME_SECONDARY = {
   primaryBgColor: "#F63D3D",
   primaryTextColor: "#fff",
   secondaryBgColor: "#000",
   // default secondaryTextColor is "#fff"

   hoverType: "changeColor",
   clickType: "rubberBand",

   onClick: () => alert("Cancelled!"),
};

// ----- ComponentFile.js -----
import ClickMe from "react-clickme";

export default function App() {
   const clickmeProps = {
      hoverType: "changeColor",
      idleType: "pulse",
      clickType: "jello",
   };

   return <ClickMe {...clickmeProps}>Example text</ClickMe>;
}`,
  },
];
